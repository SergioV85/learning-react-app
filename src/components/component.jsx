import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}