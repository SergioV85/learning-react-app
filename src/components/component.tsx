import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}