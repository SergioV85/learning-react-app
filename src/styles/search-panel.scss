@import './variables';

.c-search-panel {
  margin: 40px 0;

  &__title {
    color: $primary-light;
    font-family: $main-font;
    font-size: 0.8rem;
    text-transform: uppercase;
  }
  &__input {
    background-color: $primary-dark;
    border: 0;
    border-bottom: 2px solid $light-red;
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.7rem;
    margin: 10px 0;
    padding: 10px;
    width: 100%;

    &:focus {
      outline: none;
    }
  }
  &__controls-row {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: 15px;
  }
  .c-toggle-switcher {
    &-title {
      color: $primary-grey;
      font-family: $main-font;
      font-size: 0.7rem;
      margin-right: 20px;
      text-transform: uppercase;
    }
    &-button {
      background-color: $dark-grey;
      border-radius: 2px;
      color: $primary-grey;
      font-family: $main-font;
      font-size: 0.8rem;
      margin: 0 15px;
      text-transform: uppercase;

      &.active {
        background-color: $light-red;
        color: $primary-grey;
      }
    }
  }
  &__search-button {
    background-color: $light-red;
    border: 0;
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.9rem;
    padding: 5px;
    width: 200px;
  }
}