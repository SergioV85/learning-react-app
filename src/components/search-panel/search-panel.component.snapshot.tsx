import { shallow } from 'enzyme';
import * as React from 'react';
import { SearchPanel } from './search-panel.component';

describe('Search Panel Component Snapshot', () => {
  const mockedFn = jest.fn();

  it('should renders correctly', () => {
    const searchPanelSnapshot = shallow(
      <SearchPanel
        searchType='tilte'
        search=''
        onChangeType={mockedFn}
        onInputChange={mockedFn}
        onSearchMovies={mockedFn}
      />,
    );
    expect(searchPanelSnapshot)
      .toMatchSnapshot();
  });
});
