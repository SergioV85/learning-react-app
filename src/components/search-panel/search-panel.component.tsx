import * as React from 'react';
import { ToggleSwitcher } from './../../shared-components/toggle-switcher/toggle-switcher.component';

interface ISearchPanelProps {
  searchType: string;
  search: string;
  onChangeType(): void;
  onInputChange(): void;
  onSearchMovies(): void;
}

export class SearchPanel extends React.Component<ISearchPanelProps, {}> {
  private searchToggleButtons = [
    {
      label: 'Title',
      value: 'title',
    },
    {
      label: 'Genres',
      value: 'genres',
    },
  ];

  constructor(props: ISearchPanelProps) {
    super(props);
  }

  public render() {
    return <div className='c-search-panel'>
      <h2 className='c-search-panel__title'>Find your movie</h2>
      <input
          type='text'
          className='c-search-panel__input'
          placeholder='Find your movie'
          value={this.props.search}
          onChange={this.props.onInputChange}
      />
      <div className='c-search-panel__controls-row'>
        <ToggleSwitcher
            title='Search by'
            buttonViewClass='c-search-panel__search-type-button--title'
            buttons={this.searchToggleButtons}
            currentValue={this.props.searchType}
            name='search-type'
            type='radio'
            onToggleChange={this.props.onChangeType}
        />
        <button
            className='c-search-panel__search-button'
            onClick={this.props.onSearchMovies}
        >
          Search
        </button>
      </div>
    </div>;
  }
}
