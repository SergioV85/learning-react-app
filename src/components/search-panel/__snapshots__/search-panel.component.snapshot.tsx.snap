// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Panel Component Snapshot should renders correctly 1`] = `
<div
  className="c-page__header"
>
  <div
    className="c-page__header-shadow"
  >
    <div
      className="p-2 container"
    >
      <Component
        onBackToList={[Function]}
        showBackButton={false}
      />
      <ErrorBoundary
        errorMessage="Unfortunately, we can not show search panel"
        key="search-panel-wrapper"
      >
        <div
          className="c-search-panel"
        >
          <h2
            className="c-search-panel__title"
          >
            Find your movie
          </h2>
          <input
            className="c-search-panel__input"
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "",
                  ],
                ],
              }
            }
            placeholder="Find your movie"
            type="text"
            value=""
          />
          <div
            className="c-search-panel__controls-row"
          >
            <ToggleSwitcher
              buttonViewClass="c-search-panel__search-type-button--title"
              buttons={
                Array [
                  Object {
                    "label": "Title",
                    "value": "title",
                  },
                  Object {
                    "label": "Genres",
                    "value": "genres",
                  },
                ]
              }
              currentValue="title"
              name="search-type"
              onToggleChange={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "",
                    ],
                  ],
                }
              }
              title="Search by"
              type="radio"
            />
            <button
              className="c-search-panel__search-button"
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "",
                    ],
                  ],
                }
              }
            >
              Search
            </button>
          </div>
        </div>
        ;
      </ErrorBoundary>
    </div>
  </div>
</div>
`;
